### ISSUE TITLE
Question, how do I attach programs to a debugger?

### ISSUE THREAD

**Issue by Nukley** [otya128/winevdm#979]
How would I go about attaching 16 bit programs to a modern day debugger, like x32dbg/x64dbg? I want to patch a program.

**Comment by cracyc** [comment]
If you have winevdm installed in the registry, windbg at least just works running the win16 program directly.  That might not be so helpful for patching though since it's hard to follow the win16 code logic.  The win16 verison of the watcom debugger would likely work better for that.

**Comment by Nukley** [comment]
When trying to drag the exe I want to debug onto the win16 verison of watcom debugger, I get a setup.inf error

![image](https://user-images.githubusercontent.com/26724236/117344822-d69cbb80-ae73-11eb-9269-3a540c170924.png)

**Comment by Nukley** [comment]
wait, this isn't the debugger, this is the installer, when I try running the installer by itself, it just dies

**Comment by cracyc** [comment]
The crash is heap corruption.  Until that's fixed, the other installers including the win32 include the win16 tools.

**Comment by TheAifam5** [comment]
I was able to attach with Turbo Debugger and with OpenWatcom, but when continuing execution, debugger’s menus and hotkeys does not work and can’t break execution.

I used latest WineVDM (from CI) and latest OpenWatcom from CI.

Anyone knows which version does work?

**Comment by cracyc** [comment]
This is the nature of win16.  Because of cooperative multitasking, the debugger won't be usable while the client is running. You'll find it's the same with ntvdm or in win95.  Either set a breakpoint near what you are need to look at or use windbg (which is rather difficult to use with winevdm).
